homeassistant/components/brother/sensor.py

```diff
@@ -35,47 +35,45 @@
 
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Add Brother entities from a config_entry."""
-    brother = hass.data[DOMAIN][config_entry.entry_id]
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
 
     sensors = []
 
-    name = brother.model
     device_info = {
-        "identifiers": {(DOMAIN, brother.serial)},
-        "name": brother.model,
+        "identifiers": {(DOMAIN, coordinator.data.serial)},
+        "name": coordinator.data.model,
         "manufacturer": ATTR_MANUFACTURER,
-        "model": brother.model,
-        "sw_version": brother.firmware,
+        "model": coordinator.data.model,
+        "sw_version": coordinator.data.firmware,
     }
 
     for sensor in SENSOR_TYPES:
-        if sensor in brother.data:
-            sensors.append(BrotherPrinterSensor(brother, name, sensor, device_info))
-    async_add_entities(sensors, True)
+        if sensor in coordinator.data.data:
+            sensors.append(BrotherPrinterSensor(coordinator, sensor, device_info))
+    async_add_entities(sensors, False)
 
 
 class BrotherPrinterSensor(Entity):
     """Define an Brother Printer sensor."""
 
-    def __init__(self, printer, name, kind, device_info):
+    def __init__(self, coordinator, kind, device_info):
         """Initialize."""
-        self.printer = printer
-        self._name = name
+        self._name = f"{coordinator.data.model} {SENSOR_TYPES[kind][ATTR_LABEL]}"
+        self._unique_id = f"{coordinator.data.serial.lower()}_{kind}"
         self._device_info = device_info
-        self._unique_id = f"{self.printer.serial.lower()}_{kind}"
+        self.coordinator = coordinator
         self.kind = kind
-        self._state = None
         self._attrs = {}
 
     @property
     def name(self):
         """Return the name."""
-        return f"{self._name} {SENSOR_TYPES[self.kind][ATTR_LABEL]}"
+        return self._name
 
     @property
     def state(self):
         """Return the state."""
-        return self._state
+        return self.coordinator.data.data.get(self.kind)
```

Comment:
Nice work on the test coverage.

There are a few places where this has `coordinator.data.data` can these be eliminated as it makes the code harder to read?

Reply:
@bdraco I can try to reformat data from the library in `_async_update_data` method.

Reply:
It would be cleaner to add the functionality to the library so home assistant knows the least amount of implementation details as possible.

Reply:
@bdraco OK, I will change the library.

Reply:
@bdraco Done.